main/gradio_ui/builder.py

```
@@ -0,0 +1,208 @@
+
+async def stop_trading(kill_app: bool = False, timeout: float = 2.0):
+    global trading_task
+    try:
+        if trading_task and not trading_task.done():
+            trading_task.cancel()
+            try:
+                # Wait briefly for graceful cancellation
+                await asyncio.wait_for(trading_task, timeout=timeout)
+            except Exception:
+                # Timeout or CancelledError or other — proceed to clear task
+                pass
```

Ruff (0.13.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)